device/iot-device-client/src/main/java/com/microsoft/azure/sdk/iot/device/DeviceClient.java

```diff
@@ -575,9 +575,8 @@ public void uploadToBlobAsync(String destinationBlobName, InputStream inputStrea
      * @param request The request details for getting the SAS URI, including the destination blob name.
      * @return The file upload details to be used with the Azure Storage SDK in order to upload a file from this device.
      * @throws IOException If this HTTPS request fails to send.
-     * @throws URISyntaxException If the returned sas uri cannot be constructed correctly
      */
-    public FileUploadSasUriResponse getFileUploadSasUri(FileUploadSasUriRequest request) throws IOException, URISyntaxException
```

Comment:
This would be a breaking change since this is a public method. The best we can do here is suppress the warning and note in the javadocs that URISyntaxException is never thrown